### DISCUSSION TITLE
How can I trigger a release only by a specified scope?

### DISCUSSION THREAD

**Discussion by gary-Shen** [semantic-release/semantic-release#3257] [Q&A]
`fix: something` should not trigger a release, But `fix(scope): something` should be. Not only the release notes, But the commit analyzer.

I added a `scope` field in `releaseRules` and it didn't work.

Did I miss something?

**Comment by travi** [comment]
Both of those examples will result in a release under normal circumstances. Are you receiving an error?

**Reply by gary-Shen** [reply]
Thanks for replying. And no there are no errors. Are there any relevant examples for my scenario?

```json
{
    "repositoryUrl": "https://github.com/gary-Shen/release-test.git",
    "tagFormat": "s3-browser@v${version}",
    "branches": [
      "main"
    ],
    "plugins": [
      ["@semantic-release/commit-analyzer", {
        "preset": "conventionalcommits",
        "releaseRules": [
          {"type": "feat", "release": "minor", "scope": "s3-browser"},
          {"type": "fix", "release": "patch", "scope": "s3-browser"},
          {"type": "perf", "release": "patch", "scope": "s3-browser"},
          {"type": "revert", "release": "patch", "scope": "s3-browser"},
          {"type": "docs", "release": "patch", "scope": "s3-browser"},
          {"type": "style", "release": "patch", "scope": "s3-browser"},
          {"type": "refactor", "release": "patch", "scope": "s3-browser"}
        ]
      }],
      ["@semantic-release/release-notes-generator", {
        "preset": "conventionalcommits",
        "presetConfig": {
            "types": [
                {
                    "type": "fix",
                    "scope": "s3-browser",
                    "hidden": false,
                    "section": "Bug fixes"
                },
                {
                    "type": "feat",
                    "scope": "s3-browser",
                    "hidden": false,
                    "section": "Features"
                },
                {
                    "type": "refactor",
                    "scope": "s3-browser",
                    "hidden": false,
                    "section": "Code Refactoring"
                }
            ]
        }
      }],
      ["@semantic-release/github"]
    ]
  }
```

For more context information: [release-test](https://github.com/gary-Shen/release-test?tab=readme-ov-file#release-test)

Thanks again.

**Reply by travi** [reply]
The preset you have chosen does not prevent releases when a scope is not provided. Doing so would go beyond what conventional changelog presents typically enable. To understand your options, you need to investigate what your preset enables. Semantic-release simply uses those presets.

**Comment by gary-Shen** [comment]
Well, I debugged for a while. and found out that the commit analyzer will use the default rules if custom rules does not match any release type.

I'm using a shortcut to solve my needs.

> [https://github.com/semantic-release/commit-analyzer/blob/master/index.js#L56](https://github.com/semantic-release/commit-analyzer/blob/master/index.js#L56)

``` diff
- if (isUndefined(commitReleaseType)) {
+ if (isUndefined(commitReleaseType) && !releaseRules) {
```

It works anyway.
Thanks.

**Comment by joe-p** [comment]
I created a plugin for filtering commits based on scope which may be useful to those with a similar question: https://www.npmjs.com/package/semantic-release-scope-filter

There is also an option that allows filtering out commits without a scope
